src/qt/bitcoin.cpp

```diff
@@ -376,6 +384,43 @@ int main(int argc, char *argv[])
         snapshot.DeleteSnapshot();
     }
 
+    // Check to see if the user requested to sync from 0 -- We allow on testnet.
+    if (mapArgs.count("-syncfromzero"))
+    {
+        GRC::Upgrade syncfromzero;
+
+        if (syncfromzero.SyncFromZero())
+            LogPrintf("Syncfromzero: Success");
+
+        else
+        {
+            LogPrintf("Syncfromzero: Failed to clean up blockchain data");
+
+            std::string inftext = "";
+            // Little more work then needed but we should be translation friendly imo
+            inftext.append(_("Sync from zero: Blockchain data removal was a Failure"));
+            inftext.append("\r\n\r\n");
+            inftext.append(_("Datadir: "));
+            inftext.append(GetDataDir().string());
+            inftext.append("\r\n\r\n");
+            inftext.append(_("Due to the failure to delete the blockchain data you will be required to manually delete the data before starting your wallet."));
+            inftext.append("\r\n");
+            inftext.append(_("Failure to do so will result in undefined behaviour or failure to start wallet."));
+            inftext.append("\r\n\r\n");
+            inftext.append(_("You will need to delete the following."));
+            inftext.append("\r\n\r\n");
+            inftext.append(_("Files:"));
+            inftext.append("\r\nblk000*.dat\r\n\r\n");
+            inftext.append(_("Directories:"));
+            inftext.append("\r\ntxleveldb\r\naccrual\r\n");
+
+            ThreadSafeMessageBox(inftext, _("Gridcoin"), CClientUIInterface::OK | CClientUIInterface::MODAL);
+            QMessageBox::critical(nullptr, PACKAGE_NAME, QString::fromStdString(inftext));
```

Comment:
Why two dialogs?

Reply:
i followed suit of the bitcoin previous code. one in log (terminal) and one in gui. i found it odd to probably something we can remove from other areas too (above that code area you will see)

Reply:
Ah, I see. The `ThreadSafeMessageBox()` function doesn't actually open a message box here because the main Qt application instance hasn't initialized. The older Bitcoin code was using it to do the printf/log at the same time. Something for later 🙂